client/components/notes-formatted-block/index.jsx

```diff
@@ -66,13 +79,17 @@ export const FormattedBlock = ( { content = {} } ) => {
 			return <pre>{ descent }</pre>;
 
 		case 'theme':
-			return (
-				<a href={ content.url } target="_blank" rel="noopener noreferrer">
-					<strong>
-						<em>{ descent }</em>
-					</strong>
-				</a>
-			);
+			if ( themeUri ) {
+				if ( -1 !== themeUri.indexOf( 'wordpress.com' ) ) {
+					return <a href={ `/theme/${ themeSlug }/${ siteSlug }` }>{ descent }</a>;
+				}
+				return (
+					<a href={ themeUri } target="_blank" rel="noopener noreferrer">
+						{ descent }
+					</a>
+				);
+			}
+			return descent;
```

Comment:
early-abort would keep out the nesting and I think add to the legibility here

```js
if ( ! themeUri ) {
	return descent;
}

if ( ! /wordpress\.com/.test( themeUri ) ) {
	return <a hre…>;
}

return <a hr…>;
```

Reply:
Makes sense